### PR TITLE
Wrap label over checkbox for click handle

### PR DESCRIPTION
**Pull request by dgennetten**
#### What does this PR do?
Fix issue #1293 according to accepted practice (http://stackoverflow.com/questions/6293588/how-to-create-an-html-checkbox-with-a-clickable-label)
#### Where should the reviewer start?
https://grommet.github.io/docs/check-box/examples
#### What testing has been done on this PR?
mac/pc  edge/chrome
#### How should this be manually tested?
Verify clickable labels:  https://grommet.github.io/docs/check-box/examples
#### Do the grommet docs need to be updated?
No
#### Is this change backwards compatible or is it a breaking change?
Compatible.

### CODE REVIEW

File: src/js/components/CheckBox.js [hunk 1]
```diff
@@ -54,13 +44,15 @@ export default class CheckBox extends Component {
           </svg>
         </span>
       </span>,
-      labelNode
+      label
```
Comment (alansouzati) [inline src/js/components/CheckBox.js]: I believe this change will affect the end result right? previously was a node with a `label` tag, and now it is a string. Any motivation for this change?

Comment (dgennetten) [inline src/js/components/CheckBox.js]: Now it is a label tag embedding a checkbox. See the other changes to the file (and the PR notes).

**Comment by jlevesque** [comment]
@alansouzati 
There is also some changes that has to be consider in Radiobutton.js. Those element are mounting with two label thus throwing an wcag2 execption where a form element needs to have only one label. Having more then one label may confuses the screen readers.

```
<label className={classes}>
     <input {...props} className={`${CLASS_ROOT}__input`}
          type="radio" />
     <span className={`${CLASS_ROOT}__control`} />
     <label htmlFor={props.id} className={`${CLASS_ROOT}__label`}>
            {label}
      </label>
</label>
```

I'm currently using react-axe, and it throw this error : "Fix all of the following:
  Form element has multiple <label> elements".

I'm suggesting to change the wapper label for div, or to remove the inner label. 

Do you want me to open an issue ?

Also, is the Grommet team have or is considering to implement a WCAG2 or a a11y validator like react-axe?

**Comment by alansouzati** [comment]
Thanks for bringing this up. Yes we should definitely fix this. Can you create a github issue?

**Pull request merged**